torquedata/torquedata/routes.py

```diff
@@ -146,6 +150,44 @@ def row(sheet_name, key, fmt):
     else:
         raise Exception("Invalid format: " + fmt)
 
+
+@app.route('/api/<sheet_name>/edit-record/<key>', methods=['POST'])
+def edit_record(sheet_name, key):
+    group = request.json.get("group")
+    wiki_key = request.json.get("wiki_key")
+    new_values = json.loads(request.json.get("new_values"))
+
+    if not is_valid_id(group, wiki_key, key, sheet_name):
+        abort(403)
+
+    for field, val in new_values.items():
+        edit = {
+            "new_value": val,
```

Comment:
Should the original message be here instead of at the top level?  Because we could have

* O_Val1
* E_Val1
* E_Val2
* O_Val2 (invalidating E_Val1 and E_Val2)
* E_Val3

When do we rewrite the new original value at the top level?

Reply:
I had the understanding that the process of re-writing the original value will involve destroying/invalidating any edits that were made to the previous value. I think @kfogel said this but I'm not 100% sure.

Reply:
Right, and I know the workflow isn't fully defined.  But I'm thinking of edits after a new base value has been uploaded, because those edits would now be valid.  So we will likely have the case where there's an edit trail with some being valid, and some being invalid, and it doesn't look like the data structure can handle that case.